.projenrc.js

```diff
@@ -1,33 +1,50 @@
-const { web } = require('projen');
+const { web } = require("projen");
 
 const project = new web.ReactTypeScriptProject({
-  defaultReleaseBranch: 'main',
-  name: 'construct-hub-webapp',
-  projenUpgradeSecret: 'CDK_AUTOMATION_GITHUB_TOKEN',
+  defaultReleaseBranch: "main",
+  name: "construct-hub-webapp",
+  projenUpgradeSecret: "CDK_AUTOMATION_GITHUB_TOKEN",
 
   // Author metadata
-  authorEmail: 'construct-ecosystem-team@amazon.com',
-  authorName: 'Amazon Web Services, Inc.',
+  authorEmail: "construct-ecosystem-team@amazon.com",
+  authorName: "Amazon Web Services, Inc.",
 
   // Repository information
-  repository: 'https://github.com/cdklabs/construct-hub-webapp.git',
+  repository: "https://github.com/cdklabs/construct-hub-webapp.git",
 
   // since this is an app project, we need to enable these explicitly
   // in order to be able to publish this as an npm module.
   releaseToNpm: true,
   releaseWorkflow: true,
   package: true,
+
+  eslint: true,
+  eslintOptions: {
+    prettier: true,
+  },
 });
 
 // synthesize project files before build
 // see https://github.com/projen/projen/issues/754
-const build = project.tasks.tryFind('build')
-build.prependExec('npx projen');
+const build = project.tasks.tryFind("build");
+build.prependExec("npx projen");
 build.spawn(project.packageTask);
 
 // npm tarball will only include the contents of the "build"
 // directory, which is the output of our static website.
-project.npmignore?.addPatterns('!/build');
-project.npmignore?.addPatterns('/public');
+project.npmignore?.addPatterns("!/build");
+project.npmignore?.addPatterns("/public");
+
+// setup linting for create-react-app specific tools
+project.eslint.addRules({
```

Comment:
Should this go in projen?

Reply:
It's kinda funky because the current eslint rules aren't really friendly to the react template and requires a lot of overrides. Prettier removes some of these though, but I'm working on a separate PR to make the default rules a bit more composable and project specific where needed.